devito/grid.py

```diff
@@ -83,11 +83,11 @@ def __init__(self, shape, extent=None, origin=None, dimensions=None,
         assert (self.dim == len(self.origin) == len(self.extent) == len(self.spacing))
         # Store or create default symbols for time and stepping dimensions
         if time_dimension is None:
-            self.time_dim = TimeDimension('time', spacing=Constant(name='dt'))
-            self.stepping_dim = SteppingDimension('t', parent=self.time_dim)
+            self.time_dim = TimeDimension(name='time', spacing=Constant(name='dt'))
+            self.stepping_dim = SteppingDimension(name='t', parent=self.time_dim)
         else:
             self.time_dim = time_dimension
-            self.stepping_dim = SteppingDimension('%s_s' % time_dimension.name,
+            self.stepping_dim = SteppingDimension(name='%s_s' % time_dimension.name,
```

Comment:
OK. Bear in mind, however, that ``grid.py`` needs more clean up in general (not in this PR).

I'm raising an exception, rather than an assert, as grids are supposed to be created by users.